Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Meter/PropertyUI/LocationDrawings.tsx

```diff
@@ -44,84 +45,128 @@ const LocationDrawings = (props: IProps) => {
     const drawingSortKey = useAppSelector(LocationDrawingSlice.SortField);
     const drawingAscending = useAppSelector(LocationDrawingSlice.Ascending);
 
+    const [selectedLocation, setSelectedLocation] = React.useState<number>();
     const [showDrawings, setShowDrawings] = React.useState<boolean>(false);
-    const [hover, setHover] = React.useState<'none' | 'drawings'>('none');
+    const [showDropdown, setShowDropdown] = React.useState<boolean>();
 
     React.useEffect(() => {
-        if (drawingStatus == 'unintiated' || drawingStatus == 'changed' || drawingParentID != props.LocationID)
-            dispatch(LocationDrawingSlice.Fetch(props.LocationID));
-    }, [drawingStatus, drawingParentID, props.LocationID]);
+        if (drawingStatus == 'unintiated' || drawingStatus == 'changed' || drawingParentID != selectedLocation) {
+            dispatch(LocationDrawingSlice.Fetch(selectedLocation));
+        }
+    }, [drawingStatus, drawingParentID, selectedLocation]);
+
+    React.useEffect(() => {
+        setShowDropdown(props.Locations.length > 1);
+    }, [props.Locations])
+
+    function dropdownOptions(): dropdownOption[] {
+        const options: dropdownOption[] = [];
+        const labels: string[] = props.Locations.map(loc => loc.Name);
+        labels.forEach((label, index) => {
+            options.push({
+                Label: label,
+                Disabled: false,
+                Callback: () => {
+                    setSelectedLocation(props.Locations[index].ID);
+                    setShowDrawings(true);
+                }
+            });
+        });
+        return options;
+    }
 
     return (
         <div>
-            <button
-                type="button"
-                className={"btn btn-primary" + ((props.LocationID == null || props.LocationID == 0 || drawingData.length == 0) ? ' disabled' : '')}
-                data-tooltip={guid.current} onMouseEnter={() => setHover('drawings')} onMouseLeave={() => setHover('none')}
-                onClick={() => {
-                    if (props.LocationID != null && props.LocationID != 0 && drawingData.length != 0)
+            {showDropdown ?
+                <BtnDropdown
+                    Label={"Open Drawings " + props.Locations[0].Name}
+                    Callback={() => {
+                        setSelectedLocation(props.Locations[0].ID);
+                        setShowDrawings(true); // does this run after the selected location
+                    }}
+                    Options={dropdownOptions()}
+                />
+                : <button
+                    type="button"
+                    className={"btn btn-primary"}
```

Comment:
what happens if there are no Drawings?
what if there is no Location? 
Shouldn't this be disabled?

Reply:
Addressed in most recent push